plookup/src/table/hash_tables/tables.rs

```diff
@@ -0,0 +1,207 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) DUSK NETWORK. All rights reserved.
+
+///! In circuit hashing can be conducted using table lookups from the
+///! tables defined in this file.
+/// This currently assumes that T_S = 4. It is possible that in future T_S > 4,
+/// in which case this file will have to be adjusted. T_S is the arity of the
+/// vectors.
+use crate::table::hash_tables::constants::{N, S, T_S, V};
+use dusk_plonk::fft::Polynomial;
+use dusk_plonk::prelude::BlsScalar;
+use std::convert::TryInto;
+
+/// A HashTable consists of three different segments, each of arity
+/// vectors containing Bls Scalars. These values are precomputed and
+/// stored. These are the retrieved in the lookup functions and placed
+/// into seperate structs for the proof.
+/// A vector x in (F_p)^t goes through r rounds of a round function R.
+/// The result is another vector y in (F_p)^t.
+#[derive(Debug)]
+pub struct HashTable {
+    pub first_rows: Vec<[BlsScalar; 4]>,
+    pub middle_rows: Vec<[BlsScalar; 4]>,
+    pub end_rows: Vec<[BlsScalar; 4]>,
+}
+
+impl HashTable {
+    /// Create a new hash table and makes
+    /// empty vectors for each field.
+    pub fn new() -> Self {
+        Self {
+            first_rows: vec![],
+            middle_rows: vec![],
+            end_rows: vec![],
+        }
+    }
```

Comment:
Some of these could be arrays.. Also you already know most of the values you'll place.